src/main/java/com/cburch/logisim/file/Loader.java

```diff
@@ -310,6 +326,20 @@ public LogisimFile openLogisimFile(InputStream reader) throws IOException {
   public void reload(LoadedLibrary lib) {
     LibraryManager.instance.reload(this, lib);
   }
+  
+  public boolean export(LogisimFile file, String homeDirectory) {
+    try {
+      final var mainCircFile = LineBuffer.format("{{1}}{{2}}{{3}}{{2}}{{4}}", homeDirectory, File.separator,
+          LOGISIM_CIRCUIT_DIR, getMainFile().getName());
+      final var libraryHome = String.format("%s%s%s", homeDirectory, File.separator, LOGISIM_LIBRARY_DIR);
+      final var fwrite = new FileOutputStream(mainCircFile);
+      file.write(fwrite, this, libraryHome);
+    } catch (IOException e) {
+      //TODO: give an error message to the user
```

Comment:
Please add `System.err.println()` here. That will be easier to refactor later per #1194 

* #1136